ckanext/security/fanstatic/login_ajax.js

```diff
@@ -48,6 +48,16 @@
         showMfaForm(loginState)
       },
       error: function (xhr) {
+        var jsonResponse = null;
+        try {
+          jsonResponse = JSON.parse(xhr.responseText)
+        } catch (e) {}
+
+        if (jsonResponse && jsonResponse.hasOwnProperty('mfaCodeValid')) {
+          showError('mfa')
+          return
+        }
```

Comment:
This section is a bugfix to get the correct error message displaying if the user gets the wrong 2fa code